### PR TITLE
Metadata curation

### PR DESCRIPTION
**Pull request by MikeLippincott**
metadata_curation

### CODE REVIEW

File: 0.Preprocess_Data/1.preprocessing_morphology_data.ipynb [hunk 1]
```diff
@@ -1,4 +1,6 @@
 {
+    "attachments": {},
```
Comment (gwaybio) [inline 0.Preprocess_Data/1.preprocessing_morphology_data.ipynb:2]: Looks like this notebook needs to be rerun?

File: 10.IDR_metadata/notebooks/run_preprocessing_and_aggregation.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
```
Comment (gwaybio) [inline 10.IDR_metadata/notebooks/run_preprocessing_and_aggregation.sh:1]: I am guessing this is slurm processing instructions - it would be ideal to add a brief README within the `10.IDR_metadata` repo clarifying what this module does and how it does it.

File: 10.IDR_metadata/scripts/0.preprocess_profiles.py [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+
+# In[4]:
+
+
+# Define inputs
+feature_file = pathlib.Path(f"../../data/{cell_type}_sc_norm.parquet")
+feature_df = pd.read_parquet(feature_file)
+
+
+# In[ ]:
```
Comment (gwaybio) [inline 10.IDR_metadata/scripts/0.preprocess_profiles.py:37]: It looks like the previous cell included a keyboard interrupt. Is this expected?

The first cell of this notebook (markdown documentation) would be the place to describe if it is expected. I suspect that because you're using `papermill` we shouldn't see any cells executed.

This documentation would be super helpful to be included in the first cell and also the module-specific README.

File: 10.IDR_metadata/scripts/0.preprocess_profiles.py [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+    )
+
+
+# In[ ]:
+
+
+len(feature_df["Metadata_Well"].unique())
+
+
+# #### Combine Inducer1 and Inducer2 into one column
```
Comment (gwaybio) [inline 10.IDR_metadata/scripts/0.preprocess_profiles.py:139]: Why is this necessary? (please answer this question in the documentation)

File: 10.IDR_metadata/scripts/1.data_aggregation.py [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+# In[1]:
+
+
+# Parameters
+cell_type = "SHSY5Y"
+aggregation = True
+nomic = True
+
+
+# This noteboook aggregates the data from the previous notebooks and creates the final dataset for the analysis barring the data are aggregated in the analysis.
```
[the remark this answers is not in the document]
Comment (gwaybio) [reply 10.IDR_metadata/scripts/1.data_aggregation.py:13]: Maybe you mean "curate" would it be more clear if the file name was "1.data_curation.py"?

File: 10.IDR_metadata/scripts/2.create_processed.py [hunk 6]
```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
```
[the remark this answers is not in the document]
Comment (gwaybio) [reply 10.IDR_metadata/scripts/2.create_processed.py:4]: this will help to understand what you mean by "processed" (in the filename)

File: 10.IDR_metadata/scripts/3.create_library.py [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+import pandas as pd
+
+# In[3]:
+
+
+# set the ouput path for the library file
+output_path = pathlib.Path("../IDR_metadata/screenA/idr0000-screenA-library.txt")
+
+
+# In[5]:
```
Comment (gwaybio) [inline 10.IDR_metadata/scripts/3.create_library.py:18]: These notebooks are not executed sequentially

File: 10.IDR_metadata/scripts/3.create_library.py [hunk 8]
```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[2]:
```
Comment (gwaybio) [inline 10.IDR_metadata/scripts/3.create_library.py:4]: Please add documentation

File: 10.IDR_metadata/scripts/create_processed.py [hunk 9]
```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env python
```
Comment (gwaybio) [inline 10.IDR_metadata/scripts/create_processed.py:1]: This looks like a duplicate file that needs to be deleted (there is no number prefix)

File: figures/S9.supplemental_figure9/scripts/S9.r [hunk 10]
```diff
@@ -203,6 +205,7 @@ fig5_probabilities <- (
     + ridge_plot_apoptosis
     + ridge_plot_pyroptosis
     + plot_layout(design = layout)
+    + plot_annotation(tag_levels = "A") &  theme(plot.tag = element_text(size = 20))
```
Comment (gwaybio) [inline figures/S9.supplemental_figure9/scripts/S9.r:208]: looks like you need to remove this line (duplicated in line 209)

File: figures/S9.supplemental_figure9/scripts/S9.r [hunk 11]
```diff
@@ -175,6 +175,8 @@ ridge_plot_pyroptosis <- (
     # remove x axis label
     + theme(axis.title.x = element_blank())
     # add vertical line at 1
+    # change legend label order
+    + guides(fill = guide_legend(reverse = TRUE))
```
Comment (gwaybio) [inline figures/S9.supplemental_figure9/scripts/S9.r:179]: I don't think this had any effect.

Maybe need to rerun?

**Review by gwaybio — CHANGES_REQUESTED** [review]
Thanks for adding this IDR curation to the repo! I make several specific comments throughout, but in general, the code needs more documentation (module README and documentation cells in notebooks). The existing documentation and file names are also unclear and could be improved.

File: figures/S9.supplemental_figure9/scripts/S9.r (hunk 10 above)
Comment (MikeLippincott) [reply figures/S9.supplemental_figure9/scripts/S9.r:208]: Wrong branch...

File: 10.IDR_metadata/scripts/0.preprocess_profiles.py [hunk 12]
```diff
@@ -0,0 +1,284 @@
+#!/usr/bin/env python
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 10.IDR_metadata/scripts/0.preprocess_profiles.py:1]: Addressed

File: 10.IDR_metadata/scripts/0.preprocess_profiles.py (hunk 3 above)
Comment (MikeLippincott) [reply 10.IDR_metadata/scripts/0.preprocess_profiles.py:37]: Addressed

File: 10.IDR_metadata/scripts/0.preprocess_profiles.py (hunk 4 above)
Comment (MikeLippincott) [reply 10.IDR_metadata/scripts/0.preprocess_profiles.py:139]: Removed at this line, but updated documentation in a future line that uses this code.

File: 10.IDR_metadata/scripts/0.preprocess_profiles.py [hunk 13]
```diff
@@ -0,0 +1,284 @@
...
+        + feature_df["Metadata_inducer2_concentration"].astype(str)
+        + "_"
+        + feature_df["Metadata_inducer2_concentration_unit"].astype(str)
+    ),
+]
+feature_df["Metadata_Dose"] = np.select(condlist=conditions, choicelist=results)
+
+
+# ## N Beta Column condition generation
+# columns generated to used for linear modeling where terms separated by '__' will be a beta coefficient
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 10.IDR_metadata/scripts/0.preprocess_profiles.py:182]: Removed

File: 10.IDR_metadata/scripts/1.data_aggregation.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 10.IDR_metadata/scripts/1.data_aggregation.py:13]: Updated names and documentation

File: 10.IDR_metadata/scripts/3.create_library.py [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+    "Channels"
+] = "Hoechst 33342:nucleus;concanavalin A (con A) PhenoFluor488 conjugate:endoplasmic reticulumn;wheat germ agglutinin (WGA) PhenoFluor555 conjugate:Golgi apparatus and plasma membrane;phalloidin PhenoFluor568 conjugate: F-actin;Mitochondrial stain PhenoVue 641: mitochondria; AlexaFluor514: Cleaved N-terminal Gasdermin D"
+df.head()
+
+
+# In[6]:
+
+
+# create a dictionary to map the compound names to the compound IDs
+compund_dicts = {
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 10.IDR_metadata/scripts/3.create_library.py:47]: I am keeping this internal for now as I do not need this information outside of this notebook!

File: 10.IDR_metadata/scripts/3.create_library.py (hunk 7 above)
Comment (MikeLippincott) [reply 10.IDR_metadata/scripts/3.create_library.py:18]: re-ran this notebook to be run sequentially! :D

File: 10.IDR_metadata/notebooks/run_preprocessing_and_aggregation.sh (hunk 2 above)
Comment (MikeLippincott) [reply 10.IDR_metadata/notebooks/run_preprocessing_and_aggregation.sh:1]: addressed

**Review by gwaybio — APPROVED** [review]
Nice improvements, only nitpicks remain - feel free to merge when you are happy with it!

File: 10.IDR_metadata/scripts/0.preprocess_profiles.py [hunk 15]
```diff
@@ -0,0 +1,242 @@
...
+# coding: utf-8
+
+# This notebooks curates and preprocesses the data for IDR.
+# Specifically, it preprocesses the normalized, non-feature selected data.
+
+# In[1]:
+
+
+# Parameters
+cell_type = "PBMC"
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 10.IDR_metadata/scripts/0.preprocess_profiles.py:11]: Papermill runs for both PBMC and SH-SH5Y cells

File: 10.IDR_metadata/README.md [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+These data will be uploaded to IDR with the raw image data.
+
+## Usage
+There are four main notebooks in this module:
+1. `0.preprocess_profiles.ipynb`: This notebooks curates and preprocesses the data for IDR.
+Specifically, it preprocesses the normalized, non-feature selected data.
+2. `1.data_curation.ipynb`: This noteboook aggregates the data from the previous notebooks and creates the final aggregated dataset.
+Here we curate the data for IDR.
+3. `2.create_processed.ipynb`: This notebook creates the [processed](IDR_metadata/screenA/idr0000-screenA-processed.txt) metadata file for IDR.
+This file will include the aggregated Image-based profiles, the nELISA profiles, and the metadata for the samples.
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 10.IDR_metadata/README.md:12]: Preprocessing is cleaning the files and getting the non feature selected profiles with metadata. Curating is merging cell types and data modes.
